File: local-claude-ollama/examples/javascript/basic_chat.mjs
/**
 * Basic chat example using Anthropic JavaScript SDK with Ollama backend.
 *
 * Prerequisites:
 *   npm install @anthropic-ai/sdk
 *
 * Usage:
 *   node basic_chat.mjs
 */

import Anthropic from "@anthropic-ai/sdk";

const client = new Anthropic({
  baseURL: "http://localhost:11434",
  apiKey: "ollama", // Any value works, Ollama doesn't validate
});

async function main() {
  const message = await client.messages.create({
    model: "qwen3-coder", // Or "glm-4.7-flash" if you have it
    max_tokens: 1024,
    messages: [
      {
        role: "user",
        content: "Write a JavaScript function to check if a string is a palindrome.",
      },
    ],
  });

  console.log("Response:");
  console.log("-".repeat(40));

  for (const block of message.content) {
    if (block.type === "text") {
      console.log(block.text);
    }
  }

  console.log("-".repeat(40));
  console.log(`Stop reason: ${message.stop_reason}`);
  console.log(`Input tokens: ${message.usage.input_tokens}`);
  console.log(`Output tokens: ${message.usage.output_tokens}`);
}

main().catch(console.error);

File: local-claude-ollama/examples/python/basic_chat.py
#!/usr/bin/env python3
"""
Basic chat example using Anthropic Python SDK with Ollama backend.

Prerequisites:
    pip install anthropic

Usage:
    python basic_chat.py
"""

import anthropic


def main():
    # Configure client to use Ollama's Anthropic-compatible endpoint
    client = anthropic.Anthropic(
        base_url="http://localhost:11434",
        api_key="ollama",  # Any value works, Ollama doesn't validate
    )

    # Send a simple message
    message = client.messages.create(
        model="qwen3-coder",  # Or "glm-4.7-flash" if you have it
        max_tokens=1024,
        messages=[
            {"role": "user", "content": "Write a Python function to reverse a string."}
        ],
    )

    # Print the response
    print("Response:")
    print("-" * 40)
    for block in message.content:
        if hasattr(block, "text"):
            print(block.text)
    print("-" * 40)
    print(f"Stop reason: {message.stop_reason}")
    print(f"Input tokens: {message.usage.input_tokens}")
    print(f"Output tokens: {message.usage.output_tokens}")


if __name__ == "__main__":
    main()
